torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -475,11 +477,19 @@ void ScriptModuleSerializer::writeArchive(
       [&](const at::Tensor& tensor) {
         // returns a string to use in picker.cpp as storage obj key
         if (tensor_cdata_naming_scheme) {
-          std::string string_id =
+          std::string cdata_str =
               std::to_string(reinterpret_cast<std::intptr_t>(
                   tensor.storage().unsafeGetStorageImpl()));
-          tensor_names.push_back(string_id + ".storage");
-          storage_context_.addStorage(string_id, tensor.storage());
+          uint64_t id = 0;
+          if (storage_context_.hasStorage(cdata_str)) {
```

Comment:
If we use StorageContext to assign names in `torch.package`, we don't need this branch.